pkg/grpc/marshaler.go

```diff
@@ -0,0 +1,81 @@
+package grpc
+
+import (
+	"encoding/json"
+	"io"
+	"time"
+
+	"github.com/grpc-ecosystem/grpc-gateway/v2/runtime"
+	v1 "github.com/stackrox/rox/generated/api/v1"
+	"github.com/stackrox/rox/pkg/protocompat"
+	"google.golang.org/protobuf/encoding/protojson"
+	"google.golang.org/protobuf/proto"
+)
+
+// Deprecated: customMarshaler is a hack for keeping backward compatibility for duration objects.
+// TODO(ROX-25678): This should be deleted in 4.8.
+type customMarshaler struct {
+	*runtime.JSONPb
+}
+
+func (c customMarshaler) Unmarshal(data []byte, v interface{}) error {
+	unmarshalError := protojson.Unmarshal(data, v.(proto.Message))
+	if unmarshalError == nil {
+		return nil
+	}
+	return c.unmarshalBackwardCompatible(data, v, unmarshalError)
+}
+
+func (c customMarshaler) unmarshalBackwardCompatible(data []byte, v interface{}, unmarshalError error) error {
+	suppressCVERequest, ok := v.(*v1.SuppressCVERequest)
+	if !ok {
+		return unmarshalError
+	}
+	goStruct := SuppressCVERequestGo{}
+	err := json.Unmarshal(data, &goStruct)
+	if err != nil {
+		log.Warnf(err.Error())
+		// We want users choose the new format.
+		return unmarshalError
+	}
+	suppressCVERequest.Cves = goStruct.CVES
+	duration, err := time.ParseDuration(goStruct.Duration)
+	if err != nil {
+		return unmarshalError
+	}
+	suppressCVERequest.Duration = protocompat.DurationProto(duration)
+	log.Warnf("DEPRECATED: The duration format only supports seconds: %q", unmarshalError)
```

Comment:
Not sure `unmarshalError` is necessary related to the duration format. Can we check the error message to detect that's the problem?

Reply:
It's only printed when the old format passes but the new one not so it's a format issue.